components/inventory.ts

```diff
@@ -67,233 +91,376 @@ export function clearInventoryCache(): void {
+        }
+
+        // If the unlockable is mastery locked, checks if its unlocked based on user location progression
+        else if (
+            (unlockableMasteryData =
+                controller.masteryService.getMasteryForUnlockable(unlockable))
+        ) {
+            const locationData =
+                controller.progressionService.getMasteryProgressionForLocation(
+                    userProfile,
+                    unlockableMasteryData.Location,
+                )
 
-            const e = entP
-            const { Id: id } = unlockContainer!.Unlockable
+            const canUnlock = locationData.Level >= unlockableMasteryData.Level
 
-            if (!e) {
-                return false
+            if (canUnlock && unlockable.Type !== "evergreenmastery") {
+                acc[0].push(unlockable)
             }
 
-            if (unlockContainer.Unlockable.Type === "evergreenmastery") {
-                return false
+            // If the unlock is an evergreen package, adds its unlockables to the list
+            if (
+                unlockable.Type === "evergreenmastery" &&
+                unlockable.Properties.Unlocks
+            )
+                for (const evergreenGearId of unlockable.Properties.Unlocks) {
+                    packagedUnlocks.set(evergreenGearId, canUnlock)
+                }
+        } else {
+            const isEvergreen =
+                unlockable.Type === "evergreenmastery" ||
+                unlockable.Subtype === "evergreen"
+            const isDeluxe = DELUXE_DATA.includes(unlockable.Id)
+
+            if (isEvergreen || isDeluxe) {
+                acc[0].push(unlockable)
+            } else {
+                /**
+                 *  List of untracked items (to award to user until they are tracked to corresponding challenges)
+                 */
+                acc[1].push(unlockable)
             }
+        }
 
-            // This way of doing entitlements is a mess, redo this! - AF
-            if (gameVersion === "h3") {
-                if (WINTERSPORTS_UNLOCKABLES.includes(id)) {
-                    return (
-                        e.includes("afa4b921503f43339c360d4b53910791") ||
-                        e.includes("84a1a6fda4fb48afbb78ee9b2addd475") || // WoA Deluxe
-                        e.includes("1829590")
-                    )
-                }
+        return acc
+    }
+}
 
-                if (EXECUTIVE_UNLOCKABLES.includes(id)) {
-                    return (
-                        e.includes("6408de14f7dc46b9a33adcf6cbc4d159") ||
-                        e.includes("afa4b921503f43339c360d4b53910791") ||
-                        e.includes("84a1a6fda4fb48afbb78ee9b2addd475") || // WoA Deluxe
-                        e.includes("1829590")
-                    )
-                }
+/**
+ * Filters allowed unlockables
+ *
+ * @param gameVersion
+ * @param entP
+ * @returns boolean
+ */
+function filterAllowedContent(gameVersion: GameVersion, entP: string[]) {
```

Comment:
Not sure if we should do it as part of this PR, but this function just feels convoluted with all the hardcoded ID's and checks. Why not just get a datastructure in a file that allows us to do the same thing and keep the actual rules/configuration out of the code itself.

Reply:
It's a part of Peacock already and prevents piracy. This code is pretty much exactly how it is on master, just in a different place. It's getting kept. There isn't really a good way to do it as:
1. Peacock **still** doesn't really properly support entitlements (and either way, they're not stored in the files).
2. We'd be adding data to existing unlockable entries. We'd rather not do that as it's then signficiantly harder to update.

The feature you suggest is out of scope for this PR in my opinion, and should be discussed and a new method be implemented separately.